### PR TITLE
Remove check_match from const_eval

### LINKED ISSUE #59378: Follow up on `const_eval.rs` `check_match` call
https://github.com/rust-lang/rust/pull/59199#discussion_r267637931

### PR DESCRIPTION
**Pull request by whitfin**
This fixes #59378. 

It seems that the `check_match` may be unnecessary, so this removes it per instructions provided in the issue. I re-ran the tests for `librustc_mir` and everything seemed fine!

### CODE REVIEW

**Review by oli-obk — COMMENTED** [review]
You need to run `./x.py test --stage 1 src/test/ui --bless` in order to update the changed diagnostics due to cyclic errors having a different backtrace now.

**Comment by whitfin** [comment]
@oli-obk got it! I'll get that pushed up ASAP.

**Comment by whitfin** [comment]
@oli-obk okay, I have updated the `stderr` for those 3 tests with the different backtrace... but `ui/issues/issue-51714.rs` looks a little weird to me. It's stopped showing a warning for `irrefutable while-let pattern`. Is this expected given the removal of `check_match`? If so I can update that, but wanted to make sure before blindly changing it.

**Comment by oli-obk** [comment]
Yes I think that warning was redundant anyway.

File: src/test/ui/issues/issue-51714.stderr [hunk 1]
```diff
@@ -22,14 +22,6 @@ error[E0572]: return statement outside of function body
 LL |     [(); return while let Some(n) = Some(0) {}];
    |          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
 
-warning: irrefutable while-let pattern
```
Comment (oli-obk) [inline src/test/ui/issues/issue-51714.stderr]: Now that I think of it.... This is actually a diagnostics improvement. This now-removed diagnostic is just plain wrong.

**Comment by oli-obk** [comment]
@bors r+

thanks!

File: src/librustc_mir/const_eval.rs [hunk 2]
```diff
@@ -615,22 +615,9 @@ pub fn const_eval_raw_provider<'a, 'tcx>(
...
-
-        // Do match-check before building MIR
-        // FIXME(#59378) check_match may have errored but we're not checking for that anymore
-        tcx.check_match(def_id);
-
-        if let hir::BodyOwnerKind::Const = tcx.hir().body_owner_kind_by_hir_id(id) {
-            tcx.mir_const_qualif(def_id);
-        }
-
-        // Do not continue into miri if typeck errors occurred; it will fail horribly
```
Comment (RalfJung) [inline src/librustc_mir/const_eval.rs]: Might be worth preserving this comment.

**Pull request merged**